File: uhrwerk-common/src/main/java/io/qimia/uhrwerk/common/model/TargetModel.java
package io.qimia.uhrwerk.common.model;

import com.google.common.base.Objects;
import java.io.Serializable;

public class TargetModel extends StateModel implements Comparable<TargetModel>, Serializable {

  private static final long serialVersionUID = -3350529307251768851L;
  String format;
  Long tableId;
  Long connectionId;
  ConnectionModel connection;
  TableModel table;

  public TargetModel(TargetModelBuilder builder) {
    super(builder.deactivatedTs);
    this.id = builder.id;
    this.tableId = builder.tableId;
    this.connectionId = builder.connectionId;
    this.connection = builder.connection;
    this.table = builder.table;
    this.format = builder.format;
  }

  public static TargetModelBuilder builder() {
    return new TargetModelBuilder();
  }

  public Long getTableId() {
    return tableId;
  }

  public void setTableId(Long tableId) {
    this.tableId = tableId;
  }

  public Long getConnectionId() {
    return connectionId;
  }

  public void setConnectionId(Long connectionId) {
    this.connectionId = connectionId;
  }

  public ConnectionModel getConnection() {
    return connection;
  }

  public void setConnection(ConnectionModel Connection) {
    this.connection = Connection;
  }

  public String getFormat() {
    return format;
  }

  public void setFormat(String format) {
    this.format = format;
  }

  @Override
  public int compareTo(TargetModel o) {
    return this.getId().compareTo(o.getId());
  }

  public TableModel getTable() {
    return table;
  }

  public void setTable(TableModel table) {
    this.table = table;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (!(o instanceof TargetModel)) {
      return false;
    }
    if (!super.equals(o)) {
      return false;
    }
    TargetModel that = (TargetModel) o;
    return Objects.equal(getTableId(), that.getTableId())
        && Objects.equal(getConnectionId(), that.getConnectionId())
        && Objects.equal(getConnection(), that.getConnection())
        && Objects.equal(getTable(), that.getTable());
  }

  @Override
  public int hashCode() {
    return Objects.hashCode(
        super.hashCode(), getId(), getTableId(), getConnectionId(), getConnection(), getTable());
  }

  @Override
  public String toString() {
    return "TargetModel{" +
        "format='" + format + '\'' +
        ", tableId=" + tableId +
        ", connectionId=" + connectionId +
        ", connection=" + connection +
        '}';
  }
}

File: uhrwerk-metastore/src/main/kotlin/io/qimia/uhrwerk/common/metastore/model/TargetModel.kt
package io.qimia.uhrwerk.common.model

import io.qimia.uhrwerk.common.metastore.model.BaseModel
import io.qimia.uhrwerk.common.metastore.model.ConnectionModel
import java.sql.Timestamp
import java.time.LocalDateTime

data class TargetModel(
    var id: Long? = null,
    var tableId: Long? = null,
    var tableKey: Long? = null,
    var connectionKey: Long? = null,
    var format: String? = null,
    var tableName: String? = null,
    var hashKey: Long? = null,
    var connection: ConnectionModel? = null,
    var description: String? = null,
    var deactivatedTs: LocalDateTime? = null,
    var createdTs: Timestamp? = null,
    var updatedTs: Timestamp? = null
) : BaseModel {
    override fun id(id: Long?) {
        this.id = id
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is TargetModel) return false
        if (id != null && other.id != null) if (id != other.id) return false
        if (tableId != other.tableId) return false
        if (tableKey != other.tableKey) return false
        if (connectionKey != other.connectionKey) return false
        if (format != other.format) return false
        if (tableName != other.tableName) return false
        if (hashKey != other.hashKey) return false
        if (description != other.description) return false
        if (deactivatedTs != other.deactivatedTs) return false
        if (createdTs != other.createdTs) return false
        if (updatedTs != other.updatedTs) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id?.hashCode() ?: 0
        result = 31 * result + (tableId?.hashCode() ?: 0)
        result = 31 * result + (tableKey?.hashCode() ?: 0)
        result = 31 * result + (connectionKey?.hashCode() ?: 0)
        result = 31 * result + (format?.hashCode() ?: 0)
        result = 31 * result + (tableName?.hashCode() ?: 0)
        result = 31 * result + (hashKey?.hashCode() ?: 0)
        result = 31 * result + (description?.hashCode() ?: 0)
        result = 31 * result + (deactivatedTs?.hashCode() ?: 0)
        result = 31 * result + (createdTs?.hashCode() ?: 0)
        result = 31 * result + (updatedTs?.hashCode() ?: 0)
        return result
    }

    override fun toString(): String {
        return "TargetModel(id=$id, tableId=$tableId, tableKey=$tableKey, connectionKey=$connectionKey, format=$format, tableName=$tableName, hashKey=$hashKey, description=$description, deactivatedTs=$deactivatedTs, createdTs=$createdTs, updatedTs=$updatedTs)"
    }

}